source/Halibut/Transport/SecureWebSocketListener.cs

```diff
@@ -203,6 +205,7 @@ async Task<bool> Authorize(HttpListenerContext context, EndPoint clientName)
             if (!isAuthorized)
             {
                 log.Write(EventType.ClientDenied, "A client at {0} connected, and attempted a message exchange, but it presented a client certificate with the thumbprint '{1}' which is not in the list of thumbprints that we trust", clientName, thumbprint);
+                unauthorizedClientConnect(clientName.ToString(), thumbprint);
```

Comment:
Instead of returning something here, what if you just re-checked `verifyClientThumbprint`?

Reply:
What if I changed the callback to a 
````Func<string, string, bool>````
That would allow us to avoid calling ````verifyClientThumbprint```` again to just do what we already probably just did in the callback (trust the thumbprint).

Reply:
Changed to func.

Reply:
This was actually part of an internal conversation - @droyad had thought he was replying to me.

What if we returned an enum instead? That was is super clear what the ramification of the return value is - I find bools in method definitions are unclear, especially when doing code reviews etc. 

Maybe something like `BlockConnection`/`TrustThisThumbprintAndAllowConnection?

Reply:
Seems a bit wordy to me, and there is already a predicate param verifyClientThumbprint as a precedent. 

Returning TrustThisThumbprintAndAllowConnection suggests that the thumbprint would be added to the trusted list as a result, but that's not the case, the listener doesn't have the ability to do that. Perhaps that could be done by passing in the list instead of a predicate, but a better option would be to pass an ITrustProvider (seem my other PR).

Reply:
Thought about this a bit more and I agree returning an enum makes sense.. and it was possible to do the trust thing by handling that in the runtime.